CMakeLists.txt

```diff
@@ -46,7 +46,8 @@ if(NOT CMAKE_DISABLE_FIND_PACKAGE_CUDA AND
   # While the documentation says that it is deprecated, FindCUDA seems the
   # only easy way to determine the cublas and cusparse libraries.
   # Hence we call it unconditionally
-  find_package(CUDA)
+  # The WellContributions kernel uses __shfl_down_sync, which was introduced in CUDA 9.0
+  find_package(CUDA 9.0)
```

Comment:
I do not think this is the correct way. It will only find CUDA compatible with version 9.0. That might be what we want (in the case that every higher version is compatible with the lower ones), but we cannot be sure. I guess we need to do:
```
find_package(CUDA);
if(CUDA_VERSION VERSION_LESS "9.0")
  set(CUDA_FOUND OFF)
endif()
```

Reply:
Merging anyway. Please open another PR.